ecl/hql/hqlexpr.cpp

```diff
@@ -10705,6 +10709,62 @@ StringBuffer &CHqlUnknown::toString(StringBuffer &ret)
 
 //==============================================================================================================
 
+CHqlMacro::CHqlMacro(node_operator _op, ITypeInfo * _type, IEclPackage * _package, IAtom * _name, IFileContents * _ownedContents)
```

Comment:
I could have extended CHqlUnknown, but macros are sufficiently different that I decided it was probably better to have a separate class.  Many of the functions are very similar.  The main change is an extra package parameter, which is retrieved later on in the lexer.